node/worker/worker.go

```diff
@@ -35,20 +35,21 @@ import (
 
 // environment is the worker's current environment and holds all of the current state information.
 type environment struct {
-	signer     types.Signer
-	ethSigner  types.Signer
-	state      *state.DB     // apply state changes here
-	gasPool    *core.GasPool // available gas used to pack transactions
-	header     *block.Header
-	txs        []*types.Transaction
-	stakingTxs []*staking.StakingTransaction
-	receipts   []*types.Receipt
-	logs       []*types.Log
-	reward     reward.Reader
-	outcxs     []*types.CXReceipt       // cross shard transaction receipts (source shard)
-	incxs      []*types.CXReceiptsProof // cross shard receipts and its proof (desitinatin shard)
-	slashes    slash.Records
-	stakeMsgs  []staking.StakeMsg
+	signer              types.Signer
+	ethSigner           types.Signer
+	state               *state.DB     // apply state changes here
+	gasPool             *core.GasPool // available gas used to pack transactions
+	header              *block.Header
+	txs                 []*types.Transaction
+	stakingTxs          []*staking.StakingTransaction
+	receipts            []*types.Receipt
+	logs                []*types.Log
+	reward              reward.Reader
+	outcxs              []*types.CXReceipt       // cross shard transaction receipts (source shard)
+	incxs               []*types.CXReceiptsProof // cross shard receipts and its proof (desitinatin shard)
+	slashes             slash.Records
+	stakeMsgs           []staking.StakeMsg
+	delegationsToRemove map[common.Address][]common.Address
```

Comment:
is delegationsToRemove in worker actually used somewhere?

Reply:
Yes, it's used when caching the processor result, in line with `stakeMsgs`